app/controllers/nonresident/CalculatorController.scala

```diff
@@ -122,6 +123,60 @@ trait CalculatorController extends BaseController {
 
     Future.successful(Ok(Json.toJson(result)))
   }
+
+  def useRebasedCalculation(rebasedValue: Option[Double], rebasedCosts: Option[Double], improvementsAft: Option[Double]): Boolean = {
+    Seq(rebasedValue, rebasedCosts, improvementsAft).forall(_.isDefined)
+  }
+
+  def useTimeApportionedCalculation(disposalDate: Option[DateTime], acquisitionDate: Option[DateTime]): Boolean = {
+    (disposalDate, acquisitionDate) match {
+      case (Some(soldDate), Some(boughtDate)) => !Date.dateAfterTaxStart(boughtDate) && Date.dateAfterTaxStart(soldDate)
+      case _ => false
+    }
+  }
+
+
+  def calculateTotalGain(disposalValue: Double,
+                         disposalCosts: Double,
+                         acquisitionValue: Double,
+                         acquisitionCosts: Double,
+                         improvements: Double,
+                         rebasedValue: Option[Double],
+                         rebasedCosts: Option[Double],
+                         disposalDate: Option[DateTime],
+                         acquisitionDate: Option[DateTime],
+                         improvementsAfterTaxStarted: Option[Double]): Action[AnyContent] = Action.async { implicit request =>
+
+    val totalImprovements = improvements + improvementsAfterTaxStarted.getOrElse(0.0)
+
+    val calculateRebasedGain = {
```

Comment:
The value represents the result of the rebased gain not the calculation so it should be named `rebasedGain`